Incorporating advice for students from Information Security

KarenVines opened the pull request:
I've tried to incorporate the advice for students that came from Information Security and customise it specifically to Docker Desktop.

KarenVines committed:
Update g-local_vce_detailed.md

Trying to incorporate the good practice from Information Security

KarenVines committed:
Add files via upload

Extra figure relating to Information Security changes

psychemedia commented on g-local_vce_detailed.md:
```diff
@@ -18,7 +18,7 @@ These instructions describe how to:
 - download the module specific Docker container image that contains all the software applications needed for the module
 
-To familiarise yourself with the installation sequence, we recommend that you read through the installation process for your host operating system and the description of testing your installation at least once before carrying out the process on your own computer.
+To familiarise yourself with the installation sequence, we recommend that you read through the installation process for your host operating system and the description of testing your installation at least once before carrying out the process on your own computer. As with all software you only install this software using the verifier installer.
```
you >should< only

using the >official< installer from the original publisher's website.

KarenVines replied:
Oops that's me missing out words again. And yes your rewording is better.

psychemedia commented on g-local_vce_detailed.md:
````diff
@@ -318,3 +318,32 @@ The following flags may also be used with the `docker run` command:
+
+```
+It is possible to update to the latest version by doing the following.
+- Click on the *Update to latest* button. This will take you to the *Softwares updates* page.
````
? Software updates ?

KarenVines replied:
Oops - yes.

psychemedia commented on g-local_vce_detailed.md:
```diff
@@ -318,3 +318,32 @@ The following flags may also be used with the `docker run` command:
+- Click on the *Update to latest* button. This will take you to the *Softwares updates* page.
+- Click on *Download update* and then *Update and restart*.
+
+Note: we strongly recommend that you **do not update the software within the VCE unless told to do so by the module team**. Software updates and upgrades occasionally introduce changes that result in undesirable software behaviour, sometimes known as ‘breaking changes’, that cannot always be predicted in advance.
```
This is a bit confusing; you have been talking about the Docker application but no you switch to the s/w inside the container on the hosted VCE.

The same should apply to not updating the s/w inside a container unless told to do so (because this can break the environment). Ideally, to update s/w inside the image, we would ship a new image (this is what happens on the hosted VCE) and then tell students to update their image and create a new container.

In passing, I note the mockery of saying:

- you should update Docker if it says an update is available
- you should update the module docker image if we tell you
- you should ignore all warnings about out of date software inside the docker container...

KarenVines replied:
You're right that it does sound inconsistent (and is inconsistent!). I was trying to get the balance between complying with the guidance from Information Security and what is 'safe' from a module perspective. Maybe the answer is to tone down what we say about updating Docker. Say, to merely 'if they want to'?

psychemedia replied:
so s/thing like: upgrading Docker often improves performance and introduces new features, as well as fixing issues, but is often not essential. If you notice that your Docker application is not working correctly, and update may fix it. (You can always check the [release notes](https://docs.docker.com/release-notes/) for *Docker Desktop* and *Dokcer Engine* to see what changes are associated with any given release.)

psychemedia commented on g-local_vce_detailed.md:
```diff
@@ -318,3 +318,32 @@ The following flags may also be used with the `docker run` command:
+## Uninstalling the Docker application
+When you no longer require the Docker application to run the VCE (or VCEs for any other module you study), it is recommend that that you uninstall the application. 
+### Microsoft Windows
+For Windows devices
```
devices:

KarenVines replied:
Yes that is better.

psychemedia commented on g-local_vce_detailed.md:
```diff
@@ -318,3 +318,32 @@ The following flags may also be used with the `docker run` command:
+- In the list of Apps & Features, then scroll down to find *Docker Desktop*
+- Click on *Docker Desktop*  and select *Uninstall*.
+### Apple macOS
+For Mac devices
```
devices:

There is also inconsistency in mac and win lists regarding capitalisation of first letter...

*Who needs editors when you have collaborative comment and review tools?! ;-)*

KarenVines replied:
Indeed these tools are helping to iron out the typos and inconsistencies. I'm certainly aware that I'm much better spotting minor issues in others stuff than my own.

psychemedia replied:
Re: spotting other's issues (others'?!;-)  - yes, me too...

psychemedia replied:
if you make the changes in the branch you have issued the PR from, the updates should appear here...

KarenVines replied:
As you can tell - I'm still learning to work in GitHub.  I have done those changes in my branch and re-done the pull request. Is that right?

psychemedia commented on g-local_vce_detailed.md:
```diff
@@ -318,3 +318,32 @@ The following flags may also be used with the `docker run` command:
+Note: we strongly recommend that you **do not update the software within the VCE unless told to do so by the module team**. Software updates and upgrades occasionally introduce changes that result in undesirable software behaviour, sometimes known as ‘breaking changes’, that cannot always be predicted in advance.
+
+## Uninstalling the Docker application
+When you no longer require the Docker application to run the VCE (or VCEs for any other module you study), it is recommend that that you uninstall the application. 
```
recommend -> recommended

I disagree with the sense of this. I think we should be encouraging students to use docker more widely... How about:

When you no longer require the Docker application to run the VCE (or VCEs for any other module you are studying), you can uninstall it from your device.

KarenVines replied:
Yes I'm happy to go with that. Particularly as students may not know if it will pop up for one of their later modules. I suspect it would make many students' lives easier if module software was generally deployed using containers.

KarenVines committed:
Update g-local_vce_detailed.md

psychemedia commented on g-local_vce_detailed.md:
```diff
@@ -330,20 +330,20 @@ Checking for updates in Docker Desktop
 
 ## Uninstalling the Docker application
-When you no longer require the Docker application to run the VCE (or VCEs for any other module you study), it is recommend that that you uninstall the application. 
+When you no longer require the Docker application to run the VCE (or VCEs for any other module you study), you can uninstall the application. 
```
you may uninstall ?

KarenVines replied:
Or even 'may wish to'. 

And if we're being polite in updating the VCE I've now gone for 'asked to do so by the module team' rather than 'told to do so by the module team'

KarenVines committed:
Update g-local_vce_detailed.md

KarenVines committed:
Merge branch 'innovationOUtside:main' into main

KarenVines committed:
Update g-introduction-core.md

Upgrading the 'hints and tips to being a section rather than subsection.
Being more consistent over the use of 'a local VCE'
Correcting the occasional typo

KarenVines committed:
Update g-compute_home.md

Adding something about the notifications that Mark posts about the Open Computing Lab.

KarenVines committed:
Add files via upload

Figure showing the notifications tab on ComputeHome - and also as luck would have it, the current status of the servers.
